mypy/checkexpr.py

```diff
@@ -1840,89 +1828,234 @@ def check_op_local(self, method: str, base_type: Type, arg: Expression,
+        unions_present = isinstance(left_type, UnionType)
+
+        def make_local_errors() -> MessageBuilder:
+            """Creates a new MessageBuilder object."""
+            local_errors = self.msg.clean_copy()
+            local_errors.disable_count = 0
+            if unions_present:
+                local_errors.disable_type_names += 1
+            return local_errors
+
+        def lookup_operator(op_name: str, base_type: Type) -> Optional[Type]:
+            """Looks up the given operator and returns the corresponding type,
+            if it exists."""
+            if not self.has_member(base_type, op_name):
+                return None
+            local_errors = make_local_errors()
+            member = analyze_member_access(
+                name=op_name,
+                typ=base_type,
+                node=context,
+                is_lvalue=False,
+                is_super=False,
+                is_operator=True,
+                builtin_type=self.named_type,
+                not_ready_callback=self.not_ready_callback,
+                msg=local_errors,
+                original_type=base_type,
+                chk=self.chk,
+            )
+            if local_errors.is_errors():
+                return None
+            else:
+                return member
+
+        def lookup_definer(typ: Instance, attr_name: str) -> Optional[str]:
+            """Returns the name of the class that contains the actual definition of attr_name.
+
+            So if class A defines foo and class B subclasses A, running
+            'get_class_defined_in(B, "foo")` would return the full name of A.
+
+            However, if B were to override and redefine foo, that method call would
+            return the full name of B instead.
+
+            If the attr name is not present in the given class or its MRO, returns None.
+            """
+            mro = typ.type.mro
+            if mro is None:
+                return None
+
+            for cls in mro:
+                if cls.names.get(attr_name):
+                    return cls.fullname()
+            return None
+
+        # If either the LHS or the RHS are Any, we can't really concluding anything
+        # about the operation since the Any type may or may not define an
+        # __op__ or __rop__ method. So, we punt and return Any instead.
+
+        if isinstance(left_type, AnyType):
+            any_type = AnyType(TypeOfAny.from_another_any, source_any=left_type)
+            return any_type, any_type
+        if isinstance(right_type, AnyType):
+            any_type = AnyType(TypeOfAny.from_another_any, source_any=right_type)
+            return any_type, any_type
+
+        rev_op_name = self.get_reverse_op_method(op_name)
+
+        # STEP 1:
+        # We start by getting the __op__ and __rop__ methods, if they exist.
+
+        # Records the method type, the base type, and the argument.
+        variants_raw = []  # type: List[Tuple[Optional[Type], Type, Expression]]
```

Comment:
I would actually skip this, and just write
```
variants_raw = [(a, b),
                (c, d)]
```
in each of three branches below. The current pattern makes an impression (this is what happened to me) that the branches are not mutually exclusive (while they are).